plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,538 @@
+        reserved_multicast_entries=dict(type="int"),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["template", "template_id"], True],
+            ["state", "query", ["template", "template_id"], True],
+            ["state", "absent", ["template", "template_id"], True],
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+        mutually_exclusive=[
+            ["template", "template_id"],
+            ["state_limit_route_map_uuid", "state_limit_route_map"],
+            ["report_policy_route_map_uuid", "report_policy_route_map"],
+            ["static_report_route_map_uuid", "static_report_route_map"],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template_identifier = get_object_identifier(module.params.get("template_id"), module.params.get("template"))
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    version3_asm = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("version3_asm"))
+    fast_leave = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("fast_leave"))
+    report_link_local_groups = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("report_link_local_groups"))
+    igmp_version = module.params.get("igmp_version")
+    group_timeout = module.params.get("group_timeout")
+    query_interval = module.params.get("query_interval")
+    query_response_interval = module.params.get("query_response_interval")
+    last_member_count = module.params.get("last_member_count")
+    last_member_response_time = module.params.get("last_member_response_time")
+    startup_query_count = module.params.get("startup_query_count")
+    startup_query_interval = module.params.get("startup_query_interval")
+    querier_timeout = module.params.get("querier_timeout")
+    robustness_variable = module.params.get("robustness_variable")
+    state_limit_route_map_uuid = module.params.get("state_limit_route_map_uuid")
+    state_limit_route_map = module.params.get("state_limit_route_map")
+    report_policy_route_map_uuid = module.params.get("report_policy_route_map_uuid")
+    report_policy_route_map = module.params.get("report_policy_route_map")
+    static_report_route_map_uuid = module.params.get("static_report_route_map_uuid")
+    static_report_route_map = module.params.get("static_report_route_map")
+    maximum_multicast_entries = module.params.get("maximum_multicast_entries")
+    reserved_multicast_entries = module.params.get("reserved_multicast_entries")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "tenant", template_identifier.get("name"), template_identifier.get("uuid"))
```

Comment:
Can you refer https://github.com/CiscoDevNet/ansible-mso/pull/615/files and make changes here?

Reply:
fixed